### PR TITLE
Added pagination and mapping in facility List

### PR DESCRIPTION
**Pull request by shashankJTG**

### CODE REVIEW

File: src/containers/FacilitiesList/index.js [hunk 1]
```diff
@@ -18,56 +22,129 @@ export function FacilitiesList(props) {
...
+        ) {
+            list.map(facility => {
+                const districts = districtsList.filter(
+                    district => district.id === facility.district
+                );
+                if (districts.length > 0) facility.district = districts[0].name;
+                const ownershipTypes = ownershipTypesList.filter(
+                    ownershipType => ownershipType.id === facility.owned_by
+                );
+                if (ownershipTypes.length > 0)
```
Comment (gauravksingh) [inline src/containers/FacilitiesList/index.js:56]: it's length is checked above with isEmpty, why check again.

File: src/containers/FacilitiesList/index.js [hunk 2]
```diff
@@ -18,56 +22,129 @@ export function FacilitiesList(props) {
...
+        ownershipTypesList
+    ) => {
+        if (
+            !_.isEmpty(list) &&
+            !_.isEmpty(districtsList) &&
+            !_.isEmpty(facilityTypesList) &&
+            !_.isEmpty(ownershipTypesList)
+        ) {
+            list.map(facility => {
+                const districts = districtsList.filter(
```
Comment (gauravksingh) [inline src/containers/FacilitiesList/index.js:49]: Maybe this filter should be before isEmpty() check.

File: src/containers/FacilitiesList/index.js [hunk 3]
```diff
@@ -18,56 +22,129 @@ export function FacilitiesList(props) {
...
+    const updateFacilityListWithNames = (
+        list,
+        districtsList,
+        facilityTypesList,
+        ownershipTypesList
+    ) => {
+        if (
+            !_.isEmpty(list) &&
+            !_.isEmpty(districtsList) &&
+            !_.isEmpty(facilityTypesList) &&
```
Comment (gauravksingh) [inline src/containers/FacilitiesList/index.js:45]: Keep this order same as the order of lists handled inside if block.

File: src/containers/FacilitiesList/index.js [hunk 4]
```diff
@@ -18,56 +22,129 @@ export function FacilitiesList(props) {
...
+    const fetchPrevFacilities = () => {
+        if (hasPrev) {
+            setOffset(offset - list.length);
+        }
+    };
 
     useEffect(() => {
-        fetchFacilityList({});
-    }, [queryParams]);
+        if (!facilityTypesList) fetchFacilityTypeList();
```
Comment (gauravksingh) [inline src/containers/FacilitiesList/index.js:90]: Always be explicit with scopes i.e use { with if. Same has to be done above.

File: src/containers/FacilitiesList/index.js (hunk 1 above)
Comment (shashankJTG) [reply src/containers/FacilitiesList/index.js:56]: That check was for ownershipTypesList and this check is for ownershiptTypes.

File: src/containers/FacilitiesList/index.js (hunk 2 above)
Comment (shashankJTG) [reply src/containers/FacilitiesList/index.js:49]: If I'm understanding you correctly, then you are trying to say that first we should check that all district Ids that are present in list is also present in districtsList?

**Pull request merged**